extensions/pkg/controller/worker/controller.go

```diff
@@ -95,6 +109,24 @@ func Add(ctx context.Context, mgr manager.Manager, args AddArgs) error {
 	return nil
 }
 
+func wantMachineWatch(isAutonomousShootCluster bool, restMapper meta.RESTMapper) (bool, error) {
+	if !isAutonomousShootCluster {
+		return true, nil
+	}
+	return machineAPIPresent(restMapper)
+}
+
+func machineAPIPresent(restMapper meta.RESTMapper) (bool, error) {
+	if _, err := restMapper.KindsFor(schema.GroupVersionResource{Group: machinev1alpha1.GroupName, Version: machinev1alpha1.SchemeGroupVersion.Version, Resource: "machines"}); err != nil {
```

Comment:
This could be simplified
```suggestion
	if _, err := restMapper.KindsFor(machinev1alpha1.SchemeGroupVersion.WithResource("machines")); err != nil {
```